docs/proposals/default-service-plans.md

```diff
@@ -0,0 +1,544 @@
+    "helm.sh/hooks": pre-install # this needs to be deleted when the broker is deleted because helm won't cleanup this itself due to the hook
+  # indicates that this is a broker-provided class
+  ownerReferences:
+  - apiVersion: servicecatalog.k8s.io/v1beta1
+  	controller: true \
+Spec: \
+  externalId: b9978372-8dac-40ac-ae65-758b4a5075a5
+  # Only need specify new values not yet included in the OSB API
+  serviceType: mysql
+  defaultProvisionParameters:
+  	location: eastus
+  	resourceGroup: default
+  	sslEnforcement: disabled
+  	firewallRules:
+  	- name: "AllowAll"
+      startIPAddress: "0.0.0.0"
+      endIPAddress: "255.255.255.255"
+```
+
+
+
+#### Author
+
+Chart authors can use service types, instead of specifying specific broker's class, plan and parameters. Optional variables can be defined so an advanced user can specify a class/plan/parameters.
+
+
+```yaml
+apiVersion: servicecatalog.k8s.io/v1beta1
+kind: ServiceInstance
+metadata:
+  name: {{ template "fullname" . }}-mysql-instance
+spec:
+  serviceType: mysql-database-and-server
+  {{- if .Values.className != "" }}
+  clusterServiceClassName: {{ .Values.className }} # e.g. mysql
+  clusterServicePlanName: {{ .Values.planName }} # e.g. ha-db
+  {{- end}}
+  {{- if .Values.parameters != "" }}
+  {{ include "toYaml" .Values.parameters }}
+  {{- end}}
+
+```
+
+
+We will need to coordinate with OSB API to define an agreed upon set of service types. Below is just a suggestion:
+
+* **{{ service-name }}-database-only**: Represents a database only (requires an existing database server Service Instance). Example: mysql-database-only
+* **{{ service-name }}-server-only**: Represents a database management system or server only. Example: mysql-server-only
+* **{{ service-name }}-database-and-server**: Represents both a database server and a logical database. Example: mysql-database-and-server
+* **{{ service-name }}**: Represents a service where there are not parent-child relationships between service instances. _Parent-child relationships are a new concept in OSB that is still being discussed._ Example: redis, rabbitmq
+
+
+#### Consumer
+
+End users should be able to install a chart on a cluster with any broker that supports this proposal, and not be required to specify any broker-specific values:
+
+
+``` console
+$ helm install azure/wordpress
+# Provision a new service of type mysql and inject it into Wordpress
+```
+
+
+If the cluster operator has created custom classes, that can be specified when installing the chart:
+
+
+``` console
+$ helm install azure/wordpress \
+    --set className=custom-mysql --set planName=ha-mysql
+```
+
+
+
+## Implementation
+
+A key goal of the implementation is to avoid going off and rewriting or requiring a long term branch or fork, etc. I'd like to see this developed using a feature flag, and making all new fields optional.
+
+
+### Service Class
+
+
+| Field | Description |
+|-------|-------------|
+| Service Type | Examples: mysql, redis|
```

Comment:
again, could happen with tags

Reply:
Sorry about the confusion! I should have circled back and commented here to reflect what Scott and I figured out offline. When he is saying that it can be done via tags, he's referring to how the broker communicates to service catalog, not the manifest used by an app.

In the app, it will be a field in the spec named `serviceType` of type string, and will need to match one of the predefined service types that hopefully is coming out of the OSB F2F this week.